pkg/tagset/factory.go

```diff
@@ -0,0 +1,97 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.Datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package tagset
+
+// The Factory type is responsible for creating new Tags instances. Its
```

Comment:
This interface can't be implemented outside of the module, since it has unexported methods. I think we should state that explicitly in the interface docs.